vm/actor/src/builtin/reward/state.rs

```diff
@@ -1,110 +1,59 @@
-        let value = reward.value.clone();
+    /// The reward to be paid in total to block producers, if exactly the expected number of them produce a block.
+    /// The actual reward total paid out depends on the number of winners in any round.
+    /// This is computed at the end of the previous epoch, and should really be called ThisEpochReward.
+    #[serde(with = "biguint_ser")]
+    pub last_per_epoch_reward: TokenAmount,
 
-        rewards.add(owner.to_bytes().into(), reward)?;
+    /// The count of epochs for which a reward has been paid.
+    /// This should equal the number of non-empty tipsets after the genesis, aka "chain height".
+    pub reward_epochs_paid: ChainEpoch,
+}
 
-        self.reward_map = rewards.root()?;
-        self.reward_total += value;
-        Ok(())
+impl State {
+    pub fn new() -> Self {
+        Self::default()
     }
 
-    /// Calculates and subtracts the total withdrawable reward for an owner.
-    #[allow(dead_code)]
-    pub(super) fn withdraw_reward<BS: BlockStore>(
-        &mut self,
-        store: &BS,
-        owner: &Address,
-        curr_epoch: ChainEpoch,
-    ) -> Result<TokenAmount, String> {
-        let mut rewards = Multimap::from_root(store, &self.reward_map)?;
-        let key = owner.to_bytes();
-
-        // Iterate rewards, accumulate total and remaining reward state
-        let mut remaining_rewards = Vec::new();
-        let mut withdrawable_sum = TokenAmount::from(0u8);
-        rewards.for_each(&key, |_, reward: &Reward| {
-            let unlocked = reward.amount_vested(curr_epoch);
-            let withdrawable = unlocked
-                .checked_sub(&reward.amount_withdrawn)
-                .ok_or(format!(
-                    "Unlocked amount {} less than amount withdrawn {} at epoch {}",
-                    unlocked, reward.amount_withdrawn, curr_epoch
-                ))?;
-
-            withdrawable_sum += withdrawable;
-            if unlocked < reward.value {
-                remaining_rewards.push(Reward {
-                    vesting_function: reward.vesting_function,
-                    start_epoch: reward.start_epoch,
-                    end_epoch: reward.end_epoch,
-                    value: reward.value.clone(),
-                    amount_withdrawn: unlocked,
-                });
-            }
-            Ok(())
-        })?;
-
-        assert!(
-            withdrawable_sum < self.reward_total,
-            "withdrawable amount cannot exceed previous total"
-        );
-
-        // Regenerate amt for multimap with updated rewards
-        rewards.remove_all(&key)?;
-        for rew in remaining_rewards {
-            rewards.add(key.clone().into(), rew)?;
-        }
-
-        // Update rewards multimap root and total
-        self.reward_map = rewards.root()?;
-        self.reward_total -= &withdrawable_sum;
-        Ok(withdrawable_sum)
+    pub(super) fn get_effective_network_time(
```

Comment:
I don't think it matters but `get_effective_network_time` looks to exist in the actor implementation rather than the state.

Reply:
it does, but I chose not to follow here, as it's much cleaner this way

Reply:
made 0 sense to me why they had it the way they did